client-impl/mapreduce-client/src/main/java/uk/gov/gchq/palisade/mapreduce/PalisadeRecordReader.java

```diff
@@ -207,7 +208,11 @@ private void setupItemStream() {
         final ConnectionDetail conDetails = entry.getValue();
         final DataService service = conDetails.createService();
         //lodge request with the data service
-        final CompletableFuture<ReadResponse> futureResponse = service.read(new ReadRequest().requestId(resourceDetails.getRequestId()).resource(resource));
+        final String uuid = UUID.randomUUID().toString();
```

Comment:
The record reader should not be creating a new UUID, it should be getting it from the resourceDetails which is of type DataRequestResponse (resourceDetails probably wants renaming).

Reply:
done.